babi/file.py

```diff
@@ -673,39 +673,53 @@ def sort_selection(self, margin: Margin, reverse: bool = False) -> None:
     def _is_commented(self, lineno: int, prefix: str) -> bool:
         return self.buf[lineno].lstrip().startswith(prefix)
 
-    def _comment_remove(self, lineno: int, prefix: str) -> None:
-        line = self.buf[lineno]
+    def _start_space_offset(self, line: str) -> Tuple[int, str]:
         ws_match = WS_RE.match(line)
         assert ws_match is not None
-        ws_len = len(ws_match[0])
+        return len(ws_match[0]), ws_match[0]
+
+    def _minium_start_offset_for_selection(self) -> int:
+        s_y, e_y = self._selection_lines()
+        lengths = [
+            self._start_space_offset(self.buf[lineno])[0]
+            for lineno in range(s_y, e_y)
+        ]
+        return min(lengths)
+
+    def _comment_remove(self, lineno: int, prefix: str) -> None:
+        line = self.buf[lineno]
+        ws_len, ws_match = self._start_space_offset(line)
         rest_offset = ws_len + len(prefix)
-        if line.startswith(prefix, ws_len):
-            self.buf[lineno] = f'{ws_match[0]}{line[rest_offset:].lstrip()}'
+        if line.startswith(f'{prefix} ', ws_len):
+            self.buf[lineno] = f'{line[rest_offset + 1:]}'
+        elif line.startswith(prefix, ws_len):
+            self.buf[lineno] = f'{line[rest_offset:]}'
 
-    def _comment_add(self, lineno: int, prefix: str) -> None:
+    def _comment_add(self, lineno: int, prefix: str, s_offset: int) -> None:
         line = self.buf[lineno]
-        ws_match = WS_RE.match(line)
-        assert ws_match is not None
-        ws_len = len(ws_match[0])
-        self.buf[lineno] = f'{ws_match[0]}{prefix} {line[ws_len:]}'
+        ws_len, _ = self._start_space_offset(line)
+        self.buf[lineno] = f'{" " * s_offset}{prefix} {line}'
```

Comment:
this breaks tab indents

Reply:
Yes, oops... I didn't think about tabs...